src/main/java/frc/robot/commands/Align.java

```diff
@@ -0,0 +1,91 @@
+package frc.robot.commands;
+
+import java.util.Optional;
+
+import edu.wpi.first.math.Pair;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Pose3d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import edu.wpi.first.wpilibj2.command.ParallelCommandGroup;
+import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
+import frc.robot.Robot;
+import frc.robot.subsystems.Drivetrain;
+import frc.robot.util.Vision;
+
+public class Align extends CommandBase{
+  private Drivetrain m_drive;
+  private double setpoint;
+  private double angle;
+  // private PIDController m_pid = new PIDController(1, 0.01, 0.1);
+
+  /**
+   * Aligns the robot to a specific angle
+   * @param angle The angle to go to
+   * @param drive The drivetrain
+   */
+  public Align(double angle, Drivetrain drive){
+    addRequirements(drive);
+    setpoint=angle;
+    m_drive=drive;
+  }
+
+  private double getAngle(){
+    Optional<Pair<Pose3d, Double>> p = Vision.getEstimatedGlobalPose(m_drive.getPose());
+    if(p.isPresent() && p.get().getFirst() != null && p.get().getSecond() != null && p.get().getFirst().getX() > -10000 && p.get().getSecond() >= 0){
+      return p.get().getFirst().toPose2d().getRotation().getRadians();
+    }
+    return angle;
+  }
+
+  /**
+   * Initializes the command
+   */
+  @Override
+  public void initialize(){
+    angle=setpoint-2*Math.PI;
+    angle=getAngle();
+  }
+
+  /**
+   * Turns the robot
+   * PID broke the classbot, so this doesn't use it
+   */
+  @Override
+  public void execute(){
+    angle=getAngle();
+    double a = angle;
+    if(a<0&&setpoint>0){
```

Comment:
Use MathUtil.angleModulus for this. The way you programmed it doesn't guarantee the angle will be the closest.

Reply:
If the angle is only those values it will technically work but it can introduce a bug if the angle is a different range. Additionally, it is clearer and uses 4 less lines. 
To do this, just do this: MathUtil.angleModulus(double radians). The parameter is the angle in radians, it will output it in radians from -pi to pi.

Reply:
This is already always between -pi and pi. This code is so it knows -179 degrees is close to 180 degrees, and it should not turn 359 degrees to get to it.